lib/puppet/type/cisco_interface.rb

```diff
@@ -698,6 +724,208 @@ def is_to_s(value)
+    newvalues(
+      :true,
+      :false,
+      :default)
+  end # property switchport_mode_private_vlan_trunk_secondary
+
+  newproperty(:switchport_private_vlan_association_trunk, array_matching: :all) do
+    format = '["primary_vlan", "secondary_vlan"]'
+    desc "An array of #{format} pairs. "\
+         "Valid values match format #{format}. "\
+         'primary_vlan and secondary_vlan are integers.'
+    match_error = "Input must be of #{format}. "\
+                  'primary_vlan and secondary_vlan must be specified as integers. '\
+                  "Ex ['10', '20']"
+
+    validate do |value|
+      fail "Vlan '#{value}' #{match_error}" unless
+             /^(\s*\d+\s*)$/.match(value).to_s == value ||
+             value == 'default' || value == :default
+    end
+
+    munge do |value|
+      value == 'default' ? :default : value.to_s.gsub(/\s+/, '')
+    end
+
+    def insync?(is)
+      return true if should == [:default] && is == [:default]
+      pair = should.join(' ')
+      is.include? pair
+    end
+  end # switchport_private_vlan_association_trunk
+
+  newproperty(:switchport_private_vlan_mapping_trunk, array_matching: :all) do
+    format = '["primary_vlan", "secondary_vlan"]'
+    desc "An array of #{format} pairs. "\
+         "Valid values match format #{format}. "\
+         'primary_vlan and secondary_vlan are integers.'
+
+    match_error = "Input must be of format #{format}. "\
+                  'primary_vlan and secondary_vlan must be specified as integers.'\
+                  " Ex ['10', '20'], ['10', '20-30']"\
+                  " or ['10', '20,24']"
+
+    validate do |value|
+      fail "Vlan '#{value}' #{match_error}" unless
+           /^(\s*\d+\s*[-,\d\s]*\d+\s*)$/.match(value).to_s == value ||
+           value == 'default' || value == :default
+    end
+
+    munge do |value|
+      value == 'default' ? :default : value.to_s.gsub(/\s+/, '')
+    end
+
+    def insync?(is)
+      return true if should == [:default] && is == [:default]
+      pair = should.join(' ')
+      is.include? pair
+    end
+  end # switchport_private_vlan_mapping_trunk
+
+  newproperty(:switchport_private_vlan_trunk_allowed_vlan, array_matching: :all) do
+    format = '["vlans"]'
+    desc "An array of #{format}. "\
+         "Valid values match format #{format} with vlans as integers."
+    match_error = "must be of format #{format}. "\
+                  'vlans must be specified as integers.'\
+                  " Ex ['10'], ['20-30'] or ['20,24']"
+
+    validate do |value|
+      fail "Vlan '#{value}' #{match_error}" unless
+            value.kind_of? String
+      fail "Vlan '#{value}' #{match_error}" unless
+           /^(\s*\d+\s*[-,\d\s]*\d+\s*)$/.match(value).to_s == value ||
+           value == 'default' || value == :default
+    end
+
+    munge do |value|
+      value == 'default' ? :default : value.to_s.gsub(/\s+/, '')
+    end
+
+    def insync?(is)
+      return true if should == [:default] && is == [:default]
```

Comment:
Hi Mike.
How do we handle the case the user enters
switchport_pvlan_trunk_mapping => 'default'   
when there is no previous config?

If I do not put this return true the insync will proceed and return false since
  pair = should.join(' ')
      is.include? pair
is false.
This will break idem potency. 

Same applies to other properties.

Also how do we handle the case when 
switchport_pvlan_trunk_mapping => 'default'  and feature private vlan is not enabled?
In this case the getter will return nil. Currently I'm not handling this case. Should we?